ui/app/mixins/with-nav-to-nearest-ancestor.js

```diff
@@ -15,7 +15,7 @@ export default Mixin.create({
   navToNearestAncestor: task(function*(key) {
     let ancestors = utils.ancestorKeysForKey(key);
     let errored = false;
-    let nearest = ancestors.pop();
+    let nearest = ancestors && ancestors.pop();
```

Comment:
Hey 👋 

Just a little observation/suggestion here. I'm not even sure this is possible as I don't know how else you are using `utils.ancestorKeysForKey`.

Theoretically you could have this bug happening anywhere you are using `utils.ancestorKeysForKey` just it's not been noticed it yet, but you've only added the `ancestors` truthy check here.

_Ideally_, (although I'm not sure if its possible, and would be a bigger/riskier change). If `utils.ancestorKeysForKey` returned an empty array instead of `null` where there are no 'ancestorKeys', then your `while` here would still not run (`[].pop() === undefined`)

I just had a quick look to see where else you are using this method:

![screenshot 2019-02-11 at 09 52 16](https://user-images.githubusercontent.com/554604/52555892-db807b00-2de2-11e9-8198-698b40d87d68.png)

In `namespace-picker` you switch a `null` result back to an empty array anyway, and in `key-value-header` you do another truthy check on the return value which you could switch to a `length` check (Q: How many ancestors are there for this key? A: Zero)

So this is possible I 'think' and it's probably not that risky to do either and doing this would prevent any further bugs like this slipping in in the future.

TL;DR is You'd be changing it so the method would _always_ return an array (no alarms and no surprises), not 'mostly an array, but sometimes `null`'.

Anyway, I just had a peek here whilst drinking my morning tea, thought you might like that ^ - no matter if not.

Reply:
Thanks for pointing this out - been waayyy too long since I've looked at the util code and took a wider view of its usage. I made the necessary changes / cleanup in https://github.com/hashicorp/vault/pull/6205